### PR TITLE
docs/design: add README.md

### PR DESCRIPTION
**Pull request by rfratto**
Add README.md which outlines the proposal process, with its content copy/pasted from the approved process.

### CODE REVIEW

File: docs/design/README.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+  be kept open and the discussion continued.
+
+* **Hold**: If a discussion of a proposal requires design revisions or
+  additional information that will not be available for a couple of weeks or
+  more, the proposal review group moves the proposal to the _Hold_ state with a
+  note of what it is waiting on. Once the proposal is unblocked, the proposal
+  can be moved back to the _Active_ state for consideration during the next
+  proposal review.
+
+A [publicly available GitHub project][project] tracks proposals and their states.
```
Comment (rfratto) [inline docs/design/README.md:243]: This isn't publicly available yet, but I'm currently working with the grafana org admins to make it public.

File: docs/design/README.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+
+## Detail
+
+### Scope
+
+The proposal process is intended for user-facing features, enhancements or any
+largely scoped internal change. Other changes typically do not need a proposal,
+such as minor code refactors. If a contributor is uncertain about whether a
+proposal is needed, it is preferable to create a proposal instead of jumping
+into a change that may be declined.
```
Comment (tpaschalis) [inline docs/design/README.md:72]: minor nit: replicated paragraph.

**Review by tpaschalis — APPROVED** [review]
LGTM

File: docs/design/README.md (hunk 2 above)
Comment (rfratto) [reply docs/design/README.md:72]: Yeah, that was intentional; ideally most readers could stop reading before the detail section. Do you think I should rephrase this first instance so it's not literally a copy/paste of the same paragraph below?

Comment (tpaschalis) [reply docs/design/README.md:72]: No, it's not a big deal, let's leave it as-is.

**Comment by rfratto** [comment]
I'm going to merge regardless of whether the project board is public yet, it'll be made public as soon possible.

**Pull request merged**